exporter/datadogexporter/integrationtest/integration_test.go

```diff
@@ -39,72 +40,144 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/datadogconnector"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/testutil"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor"
 )
 
 func TestIntegration(t *testing.T) {
-	// 1. Set up mock Datadog server
-	// See also https://github.com/DataDog/datadog-agent/blob/49c16e0d4deab396626238fa1d572b684475a53f/cmd/trace-agent/test/backend.go
-	apmstatsRec := &testutil.HTTPRequestRecorderWithChan{Pattern: testutil.APMStatsEndpoint, ReqChan: make(chan []byte)}
-	tracesRec := &testutil.HTTPRequestRecorderWithChan{Pattern: testutil.TraceEndpoint, ReqChan: make(chan []byte)}
-	server := testutil.DatadogServerMock(apmstatsRec.HandlerFunc, tracesRec.HandlerFunc)
-	defer server.Close()
-
-	// 2. Start in-process collector
-	factories := getIntegrationTestComponents(t)
-	app, confFilePath := getIntegrationTestCollector(t, server.URL, factories)
-	go func() {
-		assert.NoError(t, app.Run(context.Background()))
-	}()
-	defer app.Shutdown()
-	defer os.Remove(confFilePath)
-	waitForReadiness(app)
-
-	// 3. Generate and send traces
-	sendTraces(t)
-
-	// 4. Validate traces and APM stats from the mock server
-	var spans []*pb.Span
-	var stats []*pb.ClientGroupedStats
-
-	// 5 sampled spans + APM stats on 10 spans are sent to datadog exporter
-	for len(spans) < 5 || len(stats) < 10 {
-		select {
-		case tracesBytes := <-tracesRec.ReqChan:
-			gz := getGzipReader(t, tracesBytes)
-			slurp, err := io.ReadAll(gz)
-			require.NoError(t, err)
-			var traces pb.AgentPayload
-			require.NoError(t, proto.Unmarshal(slurp, &traces))
-			for _, tps := range traces.TracerPayloads {
-				for _, chunks := range tps.Chunks {
-					spans = append(spans, chunks.Spans...)
-					for _, span := range chunks.Spans {
-						assert.Equal(t, span.Meta["_dd.stats_computed"], "true")
+	t.Run("with feature gate enabled", func(t *testing.T) {
```

Comment:
Seems lots of duplicates here, can we refactor the 2 test cases to something like:
```
for _, tt := range []struct {
		name  string
		enableFG bool
	}
...
```